src/main/java/org/fcrepo/client/BodyRequestBuilder.java

```diff
@@ -244,4 +245,17 @@ protected BodyRequestBuilder linkAcl(final String aclUri) {
         }
         return this;
     }
+
+    /**
+     * Provide a value for the if-state-token header for this request.
+     *
+     * @param token state token value
+     * @return this builder
+     */
+    protected BodyRequestBuilder ifStateToken(final String token) {
```

Comment:
`BodyRequestBuilder` is extended by:
- `HistoricMementoBuilder`
- `PostBuilder`
as well as
- `PutBuilder`
- `PatchBuilder`

Since `X-If-State-Token` is not available for POST and Mementos, should we be preventing its use in those cases?

Reply:
It is a protected method and only exposed if the child classes override it with a public version of the method, so it should be okay from the perspective of users.